### PR TITLE
Sanity schema settings. Sanity singleton documents

### PR DESCRIPTION
**Pull request by smovzh**

### CODE REVIEW

File: sanity.config.ts [hunk 1]
```diff
@@ -40,6 +46,14 @@ export default defineConfig({
   // edit schemas in './src/schemas'
   schema: {
     types: schemaTypes,
+    templates: (templates) =>
+      templates.filter(({ schemaType }) => !singletonTypes.has(schemaType)),
+  },
```
Comment (smovzh) [inline sanity.config.ts:51]: I found this in Sanity documentation but I don't know why we need that change to templates.
It works even without this change.

File: sanity.config.ts [hunk 2]
```diff
@@ -40,6 +46,14 @@ export default defineConfig({
...
+  document: {
+    actions: (input, context) =>
+      singletonTypes.has(context.schemaType)
+        ? input.filter(({ action }) => action && singletonActions.has(action))
+        : input,
```
Comment (smovzh) [inline sanity.config.ts:56]: Allow only predefined actions for the singleton documents (publish, discard, restore)

File: src/schemas/settings.ts
Comment (smovzh) [inline src/schemas/settings.ts:1]: It should be changed by our needs. I added several fields just for example.

File: src/utils/deskStructure.js [hunk 3]
```diff
@@ -22,11 +22,16 @@ export const uawitStructure = (S) =>
...
-                ['widget'].includes(listItem.getId())
-              ),
+              S.listItem()
+                .title('General Configuration')
+                .child(
+                  S.document()
+                    .title('General Configuration')
+                    .schemaType('settings')
+                    .documentId('general')
+                )
```
Comment (smovzh) [inline src/utils/deskStructure.js]: I like this!!!

`
S.document().title('General Configuration').schemaType('settings').documentId('general')
`

It shows on the right panel the document of Settings (our schema) with predefined Id (general)

So, it is possible to have several documents of Settings with predefined Ids. And content manager can edit only predefined amount of items but cannot create new ones.      
General-Configuration-Project-Name

File: src/utils/deskStructure.js [hunk 4]
```diff
@@ -22,11 +22,17 @@ export const uawitStructure = (S) =>
...
-              ),
+              S.listItem()
+                .title('General Configuration')
+                .id("general")
+                .child(
+                  S.document()
+                    .title('General Configuration')
+                    .schemaType('settings')
+                    .documentId('general')
+                )
```
Comment (smovzh) [inline src/utils/deskStructure.js:35]: I added several fields to general configuration just for example. We can extend it according to our needs.

General-Configuration-Project-Name

File: src/schemas/settings.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  fields: [
+    defineField({
+      name: 'email',
+      title: 'Email',
+      type: 'string',
+      group: 'contacts',
+    }),
+    defineField({
+      name: 'telephone',
+      title: 'Telephone',
```
Comment (julia-dizhak) [inline src/schemas/settings.ts:26]: minor: I would name it just phone

File: src/utils/deskStructure.js (hunk 4 above)
Comment (julia-dizhak) [reply src/utils/deskStructure.js:35]: yeah, probably for now just one is enough

**Review by julia-dizhak — APPROVED** [review]
thanks!

**Pull request merged**